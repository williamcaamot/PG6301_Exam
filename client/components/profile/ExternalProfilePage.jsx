import React, { useEffect, useState } from "react";
import ErrorMessage from "../globals/ErrorMessage.jsx";
import { useMatch } from "react-router-dom";
import Profile from "./Profile.jsx";

function ExternalProfilePage() {
  const [errorMessage, setErrorMessage] = useState();

  const match = useMatch("/profile/:email");
  const email = match.params.email;

  const [user, setUser] = useState();

  async function getUser() {
    try {
      const res = await fetch(`/api/v1/profile/${email}`);
      const { message, data } = await res.json();

      if (res.status !== 200) {
        setErrorMessage(message || "En ukjent feil har oppstÃ¥tt");
        return;
      }
      setUser(data);
    } catch (e) {
      setErrorMessage(e.message);
    }
  }
  useEffect(() => {
    getUser();
  }, []);

  return (
    <>
      <div className={"pageContentWrapper"}>
        <div className={"innerWrapper"}>
          <ErrorMessage message={errorMessage} />
          <Profile user={user} />
        </div>
      </div>
    </>
  );
}

export default ExternalProfilePage;
