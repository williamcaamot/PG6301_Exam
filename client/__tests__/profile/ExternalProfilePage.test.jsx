import renderer, { act } from "react-test-renderer";
import { MemoryRouter } from "react-router-dom";
import ExternalProfilePage from "../../components/profile/ExternalProfilePage.jsx";

describe("render external profile page, the page other users can see", () => {
  it("should fetch profile details and dispaly them", async () => {
    const user = {
      _id: "655dd38a4c15f00c20bc3fcf",
      bio: "dawawddawJeg er kul",
      email: "williamcoucheronaamot@gmail.com",
      family_name: "Coucheron-Aamot",
      name: "William",
      nickname: "WILLIAM123wwwww",
      picture:
        "https://lh3.googleusercontent.com/a/ACg8ocIOgSmXkWTJqoKPe5Hk3JlkkhfUNer4sWkVo3UuIa_dzg=s96-c",
    };
    let component;

    await act(async () => {
      component = renderer.create(
        <MemoryRouter initialEntries={["/profile/william@gmail.com"]}>
          <ExternalProfilePage
            getUser={() => {
              user;
            }}
          />
        </MemoryRouter>,
      );
    });
    expect(component).toMatchSnapshot();
  });
});
