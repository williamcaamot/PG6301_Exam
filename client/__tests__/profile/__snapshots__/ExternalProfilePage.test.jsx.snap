// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render external profile page, the page other users can see should fetch profile details and dispaly them 1`] = `
<div
  className="pageContentWrapper"
>
  <div
    className="innerWrapper"
  >
    <h2>
      Loading...
    </h2>
  </div>
</div>
`;
